Telegraf crashes with the following error: panic: runtime error: invalid memory address or nil pointer dereference

I was running telegraf 1.18.0-alpine on docker container on Azure App Service. 

### Relevant telegraf.conf:
```
[agent]
  interval = "20s"
  round_interval = true
  metric_batch_size = 1000
  metric_buffer_limit = 10000
  collection_jitter = "0s"
  flush_interval = "20s"
  flush_jitter = "0s"
  precision = ""
  hostname = ""
  omit_hostname = false
  debug = true


[[outputs.azure_monitor]]
  region = "northeurope"
  resource_id = "/subscriptions/*******/resourceGroups/*****/providers/Microsoft.Web/sites/****"
  namespace_prefix = "Telegraf_app/"

[[inputs.prometheus]]
  urls = ["https://*****.azurewebsites.net/metrics"]
  metric_version = 2
```

### System info:

Telegraf in docker container: version: 1.18.0-alpine

### Steps to reproduce:

launch telegraf

### Expected behavior:

Sending logs to Azure monitor

### Actual behavior:

Telegraf crashed.

### Additional info:

logs:
```
- Container for ***** site ***** is unhealthy, Stopping site.
- Stopping site ***** because it is not healthy.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x22ebdd5]
goroutine 114 [running]:
github.com/influxdata/telegraf/plugins/outputs/azure_monitor.(*AzureMonitor).send(0xc00022b760, 0xc0004f0000, 0x21c7, 0x2800, 0x0, 0x0)
/go/src/github.com/influxdata/telegraf/plugins/outputs/azure_monitor/azure_monitor.go:348 +0x335
github.com/influxdata/telegraf/plugins/outputs/azure_monitor.(*AzureMonitor).Write(0xc00022b760, 0xc0000bd900, 0x26, 0x26, 0xc0000bd900, 0xc000535ee0)
/go/src/github.com/influxdata/telegraf/plugins/outputs/azure_monitor/azure_monitor.go:325 +0x5fa
github.com/influxdata/telegraf/models.(*RunningOutput).write(0xc0000ea900, 0xc0000bd900, 0x26, 0x26, 0x26, 0xc0003e7801)
/go/src/github.com/influxdata/telegraf/models/running_output.go:246 +0x99
github.com/influxdata/telegraf/models.(*RunningOutput).Write(0xc0000ea900, 0x86d5000, 0xf6000)
/go/src/github.com/influxdata/telegraf/models/running_output.go:203 +0x1f8
github.com/influxdata/telegraf/agent.(*Agent).flushOnce.func1(0xc00049c3c0, 0xc000bb9290)
/go/src/github.com/influxdata/telegraf/agent/agent.go:843 +0x27
created by github.com/influxdata/telegraf/agent.(*Agent).flushOnce
/go/src/github.com/influxdata/telegraf/agent/agent.go:842 +0x7a
```

--- a/plugins/outputs/azure_monitor/azure_monitor.go
+++ b/plugins/outputs/azure_monitor/azure_monitor.go
@@ -193,7 +193,7 @@ func (a *AzureMonitor) Connect() error {
 
 	a.auth, err = auth.NewAuthorizerFromEnvironmentWithResource(defaultAuthResource)
 	if err != nil {
-		return nil
+		return err
 	}
 
 	a.Reset()

